File: VB/HowToAccessApplicationModelInCode.Module.Win/SelectViewController.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Public Class SelectViewController
	Inherits DevExpress.ExpressApp.ViewController

	Public Sub New()
		MyBase.New()

		'This call is required by the Component Designer.
		InitializeComponent()
		RegisterActions(components) 

   End Sub
   Protected Overrides Sub OnActivated()
      MyBase.OnActivated()
      SelectViewAction.BeginUpdate()
      SelectViewAction.Items.Clear()
      Dim node As DictionaryNode = Application.Model.RootNode.GetChildNode("Views")
      For Each dnode As DictionaryNode In node.GetChildNodes("ListView", "ClassName", _
            View.ObjectTypeInfo.Type.FullName, False)
         Dim current As ChoiceActionItem = _
            New ChoiceActionItem(dnode.GetAttributeValue("ID"), dnode)
         SelectViewAction.Items.Add(current)
         If dnode Is View.Info Then
            SelectViewAction.SelectedItem = current
         End If
      Next dnode
      SelectViewAction.EndUpdate()
   End Sub

   Private Sub SelectViewAction_Execute(ByVal sender As System.Object, ByVal e As DevExpress.ExpressApp.Actions.SingleChoiceActionExecuteEventArgs) Handles SelectViewAction.Execute
      View.SetInfo(CType(e.SelectedChoiceActionItem.Data, DictionaryNode))
   End Sub
End Class

File: CS/HowToAccessApplicationModelInCode.Module.Win/SelectViewController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Model;

namespace HowToAccessApplicationModelInCode.Module.Win {
    public partial class SelectViewController : ViewController {
        public SelectViewController() {
            InitializeComponent();
            RegisterActions(components);
        }
        private void SelectViewAction_Execute(object sender, SingleChoiceActionExecuteEventArgs e) {
            View.SetInfo((IModelObjectView)e.SelectedChoiceActionItem.Data);
        }
        protected override void OnActivated() {
            base.OnActivated();
            SelectViewAction.BeginUpdate();
            SelectViewAction.Items.Clear();
            foreach(IModelObjectView node in View.Model.Application.Views.GetNodes<IModelListView>()) {
                if(node.ModelClass == ((IModelObjectView)View.Model).ModelClass) {
                    ChoiceActionItem current = new ChoiceActionItem(node.Id, node);
                    SelectViewAction.Items.Add(current);
                    if (node == View.Model) SelectViewAction.SelectedItem = current;
                }
            }
            SelectViewAction.EndUpdate();
        }
    }
}
